web/impact/impact/v1/helpers/criterion_helper.py

```diff
@@ -32,25 +32,19 @@ class CriterionHelper(ModelHelper):
     INPUT_KEYS = ALL_KEYS
 
     specific_helpers = {}
-    helper_instances = {}
 
     @classmethod
     def register_helper(cls, klass, type, name):
         cls.specific_helpers[(type, name)] = klass
 
     @classmethod
     def find_helper(cls, criterion):
-        helper = cls.helper_instances.get((criterion.name))
-        if helper is None:
-            helper = cls.specific_helpers.get((criterion.type, criterion.name),
-                                              cls)(criterion)
-            cls.helper_instances[(criterion.name)] = helper
+        helper = cls.specific_helpers.get(
+            (criterion.type, criterion.name),
+            cls
```

Comment:
It isn't passing its own class, it's passing the field criterion.type, which is a string. I don't love this paired key - I think criterion.id should be sufficient - but since this is an urgent fix that needs to be deployed for immediate use I don't think that this is a good time to address that issue. I can make a follow-on ticket about fixing that if you like.

Reply:
I believe Shankar is referring to the get method passing `cls` as the second argument.

Shankar, that is the default value, i.e. if the key `(criterion.type, criterion.name)` does not exist, we return the current executing class to be used as the helper class. hence the `cls`

Reply:
Thank you for that clarification @VincentHokie 
@shanbady If you have specific concrete concerns about the existing code that you think need to be addressed, please raise them. We need to get this merged. 
This code smell predates these changes. I'm happy to address them in a follow-on ticket if you want, but please focus on the changes at hand so we can get this done.